openpype/hosts/maya/plugins/publish/collect_arnold_scene_source.py

```diff
@@ -23,14 +23,45 @@ def process(self, instance):
                 continue
             if objset.endswith("content_SET"):
                 members = cmds.ls(members, long=True)
+                members_list = []
+                for member in members:
+                    shape = cmds.listRelatives(
+                        member, shapes=True, fullPath=True)
+                    if not shape:
+                        continue
+                    members_list = members + shape
+                    group_name = "|{}".format(member)
+                    if group_name in members_list:
+                        members_list.remove(group_name)
```

Comment:
This will be very very slow for large lists of members.  Also, the members_list is being overridden for EACH member and the previous shape will be lost. Are you sure this is doing what you want it to?

I'm pretty sure this is not doing what you expect it to.

First of:
1. It assumes the top level instance will only **ever** contain objectSets otherwise the code will fail on the `cmds.sets()` command (not a new issue in this PR)
2. The logic for both `proxy_SET` and `content_SET` is basically the same, but just stored elsewhere. We should deduplicate the code so that it's clear they are basically doing the same thing.
3. Why do we do a per member listRelatives check for a shape, then _something?_ with a group name that I have no idea what it's based on, to then remove the top level? But directly after we still extend all children? What is that we are trying to solve?

Reply:
It seems that the updated version doesn't show up. it should look way much simpler than this.